velox/functions/lib/aggregates/tests/utils/AggregationTestBase.cpp

```diff
@@ -578,17 +569,31 @@ void AggregationTestBase::testReadFromFiles(
   }
 
   auto input = AssertQueryBuilder(builder.planNode()).copyResults(pool());
-  if (input->size() < 2) {
+  if (input->size() <= 0) {
     return;
   }
-  auto size1 = input->size() / 2;
-  auto size2 = input->size() - size1;
-  auto input1 = input->slice(0, size1);
-  auto input2 = input->slice(size1, size2);
+
   std::vector<std::shared_ptr<exec::test::TempFilePath>> files;
   std::vector<exec::Split> splits;
+  std::vector<VectorPtr> inputs;
   auto writerPool = rootPool_->addAggregateChild("AggregationTestBase.writer");
-  for (auto& vector : {input1, input2}) {
+
+  // Splits and writes the input vectors into two files, to some extent,
+  // involves shuffling of the inputs. So only split input if allowInputShuffle_
+  // is true. Otherwise, only write into a single file.
+  if (allowInputShuffle_ && input->size() >= 2) {
+    auto size1 = input->size() / 2;
+    auto size2 = input->size() - size1;
+    auto input1 = input->slice(0, size1);
+    auto input2 = input->slice(size1, size2);
+    inputs = {input1, input2};
+  } else if (input->size() > 0) {
+    inputs.push_back(input);
+  } else {
```

Comment:
Do we have a case when the code goes into the 'else' branch?

Reply:
Removed. We should never reach here.